helios-client/src/main/java/com/spotify/helios/client/RetryingRequestDispatcher.java

```diff
@@ -0,0 +1,159 @@
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * A {@link RequestDispatcher} that retries. Uses {@link DefaultRequestDispatcher}
+ * as a delegate.
+ */
+class RetryingRequestDispatcher implements RequestDispatcher {
+
+  private static final Logger log = LoggerFactory.getLogger(RetryingRequestDispatcher.class);
+
+  private static final long RETRY_TIMEOUT_MILLIS = SECONDS.toMillis(60);
+  private static final long DEFAULT_DELAY = 5;
+  private static final TimeUnit DEFAULT_DELAY_TIMEUNIT = SECONDS;
+
+  private final ListeningScheduledExecutorService executorService;
+  private final RequestDispatcher delegate;
+  private final Clock clock;
+  private final long delay;
+  private final TimeUnit delayTimeUnit;
+
+  RetryingRequestDispatcher(final Supplier<List<Endpoint>> endpointSupplier,
+                            final String user,
+                            final ListeningScheduledExecutorService executorService) {
+    this(new DefaultRequestDispatcher(endpointSupplier, user, executorService), executorService,
+         new SystemClock(), DEFAULT_DELAY, DEFAULT_DELAY_TIMEUNIT);
+  }
+
+  RetryingRequestDispatcher(final RequestDispatcher delegate,
+                            final ListeningScheduledExecutorService executorService,
+                            final Clock clock,
+                            final long delay,
+                            final TimeUnit delayTimeUnit) {
+    this.delegate = delegate;
+    this.executorService = executorService;
+    this.clock = clock;
+    this.delay = delay;
+    this.delayTimeUnit = delayTimeUnit;
+  }
+
+  @Override
+  public ListenableFuture<Response> request(final URI uri,
+                                            final String method,
+                                            final byte[] entityBytes,
+                                            final Map<String, List<String>> headers) {
+    final long deadline = clock.now().getMillis() + RETRY_TIMEOUT_MILLIS;
+    final SettableFuture<Response> future = SettableFuture.create();
+    final Supplier<ListenableFuture<Response>> code = new Supplier<ListenableFuture<Response>>() {
+      @Override
+      public ListenableFuture<Response> get() {
+        return delegate.request(uri, method, entityBytes, headers);
+      }
+    };
+    startRetry(future, code, deadline, delay, delayTimeUnit, Predicates.<Response>alwaysTrue());
+    return future;
+  }
+
+  @Override
+  public void close() {
+    delegate.close();
+  }
+
+  private void startRetry(final SettableFuture<Response> future,
+                          final Supplier<ListenableFuture<Response>> code,
+                          final long deadline,
+                          final long delay,
+                          final TimeUnit timeUnit,
+                          final Predicate<Response> retryCondition) {
```

Comment:
It is. I lifted this code from github.com/spotify/futures-extra. I'll remove the predicate.